### PR TITLE
fix: fixed the venue sub menu not closing

### LINKED ISSUE #205: Venue sub menu doesn't close
When you click on the venue dropdown, it should close when you click anywhere on the page.

AsyncAPI-Conference (4)

### PR DESCRIPTION
**Pull request by ShashwatPS**
Description :
Fixed the part where the venue sub menu on the navbar was not closing when clicked anywhere but inside the menu.

Related issue(s) :

Resolves #205

### CODE REVIEW

File: components/Navbar/navbar.js [hunk 1]
```diff
@@ -12,6 +12,17 @@ function Navbar() {
 	const isTablet = useMediaQuery({ maxWidth: '1118px' });
 	const [drop, setDrop] = useState(false);
 	const [show, setShow] = useState(null);
+	useEffect(() => {
+		function handleClosing(event) {
+			if (show && !event.target.closest('.subMenu')) {
+				setShow(false);
+			}
+		}
```
Comment (AceTheCreator) [inline components/Navbar/navbar.js:20]: put this function outside the useEffect

File: components/Navbar/navbar.js [hunk 2]
```diff
@@ -12,6 +12,17 @@ function Navbar() {
...
+		document.addEventListener('mousedown', handleClosing);
+		return () => {
+			document.removeEventListener('mousedown', handleClosing);
+		};
```
Comment (AceTheCreator) [inline components/Navbar/navbar.js:24]: I don't think the `return` is necessary

File: components/Navbar/navbar.js [hunk 3]
```diff
@@ -61,7 +72,7 @@ function Navbar() {
 											)}
 										</div>
 										{show && show === link.title && link.subMenu && (
-											<div className='absolute z-[9] mt-8 w-[140px] rounded-md left-[-15px] gradient-bg pl-2 pt-1 flex flex-col justify-center space-y-0'>
+											<div className='subMenu absolute group-hover:bg-red-300 z-[9] mt-8 w-[140px] rounded-md left-[-15px] gradient-bg pl-2 pt-1 flex flex-col justify-center space-y-0'>
```
Comment (AceTheCreator) [inline components/Navbar/navbar.js:75]: why did you add this group-hover:bg-red-300

**Review by AceTheCreator — COMMENTED** [review]
Left you some review

**Review by AceTheCreator — CHANGES_REQUESTED** [review]
Left you some comments

File: components/Navbar/navbar.js (hunk 1 above)
Comment (ShashwatPS) [reply components/Navbar/navbar.js:20]: Will do this.

File: components/Navbar/navbar.js (hunk 2 above)
Comment (ShashwatPS) [reply components/Navbar/navbar.js]: True Fixed.

File: components/Navbar/navbar.js (hunk 3 above)
Comment (ShashwatPS) [reply components/Navbar/navbar.js]: Its my fault here. Sorry for the inconvenience.

**Review by ShashwatPS — COMMENTED** [review]
Done with the changes. @AceTheCreator

**Comment by ShashwatPS** [comment]
@AceTheCreator Can you review this please whenever you have the time ?

**Review by AceTheCreator — APPROVED** [review]

**Comment by AceTheCreator** [comment]
Well done @ShashwatPS

**Pull request merged**